app/models/Restaurant.java

```diff
@@ -79,6 +83,7 @@
      * @return true if validation is successful
      */
     @Override
+    @JsonView(View.AdditionalDetails.class)
     public boolean isValid() {
         return !name.equals("") && !address.getCity().equals("") && !address.getCountry().equals("") &&
                 !address.getStreetName().equals("") && coordinates.getLatitude() != 0 && coordinates.getLongitude() != 0
```

Comment:
you should use isEmpty() instead of equals("")

also, null check should be performed on all fields prior to isEmpty